### ISSUE TITLE
Paladin Rapid Smithing ( Renewal ) not using official behaviour

### ISSUE THREAD

**Issue by Paoly28** [rathena/rathena#535]
it doesn't count status atk , weapon atk and equip atk in the formula moreover it also doesn't benefit from gloomy shyness +50% atk buff and should also be melee after post balancing. 

old thread about renewal rapid smithing 
http://forums.irowiki.org/showthread.php?t=77472

2014 vid of Rapid smithing test renewal
" you can see that he failed his tests since archer skeletons didn't worked since it's melee now"
https://www.youtube.com/watch?v=pS9tO138wuE
part 2 of test
http://www.youtube.com/watch?v=BTQtvwGmGjU

PVP vid of Rapid smithing w/ patchnote on youtube desc

https://www.youtube.com/watch?v=cZZeM6Dkt3k

Photo of mine testing rapid smithing 

http://i58.tinypic.com/fbhpch.jpg

**Comment by aleos89** [comment]
Keep in mind, the bonus from Gloomy Shyness is not a solid +50% at level 5. In the notes section it states the formula as well:
Random number between [15 ~ (Voice Lesson Skill Level x 5) + (Skill Level x 10)] %

**Comment by Paoly28** [comment]
ofcourse i did the test with maxed voice lessons and lvl5 gloomy shyness and i didn't even get any bonus.

**Comment by Paoly28** [comment]
so it should actually be 15 ~ 100% today i learned ty aleos ^^

**Comment by Paoly28** [comment]
Tested it again, Gloomy works with it.

**Comment by Paoly28** [comment]
is this fixed already ? gloomy works fine but shield chain is definitely broken.

**Comment by aleos89** [comment]
Sorry about this. My mistake. Was going through reports the other day, didn't mean to close this one.

**Comment by Paoly28** [comment]
`#ifdef RENEWAL
			if (sd) {
				short index = sd->equip_index[EQI_HAND_L];

				if (index >= 0 &&
					sd->inventory_data[index] &&
					sd->inventory_data[index]->type == IT_WEAPON)
					wd.equipAtk += sd->inventory_data[index]->weight/10;

				wd = battle_calc_damage_parts(wd, src, target, skill_id, skill_lv);
			}`

adds eqAtk , wAtk and sAtk on shield chain

**Comment by Paoly28** [comment]
#ifdef RENEWAL
if (sd) {
short index = sd->equip_index[EQI_HAND_L];

            if (index >= 0 &&
                sd->inventory_data[index] &&
                sd->inventory_data[index]->type == IT_ARMOR)
                wd.equipAtk += sd->inventory_data[index]->weight/10;

            wd = battle_calc_damage_parts(wd, src, target, skill_id, skill_lv);
        }
adds shield weight as eqAtk , also adds eqAtk, sAtk and wAtk in the renewal formula

**Comment by violetharmony** [comment]
Rapid Smiting (Shield Chain) = [(Atk + ShieldWeight + (refine×4))*% - softDef - hardDef]×5

**Comment by ecdarreola** [comment]
Is the fix included in the #4072 ?

**Comment by mmo4th** [comment]
> Is the fix included in the #4072 ?

nope

**Comment by Lemongrass3110** [comment]
```
"쉴드체인 (Shield Chain)",
"MAX Lv : 5",
"^777777습득조건 : 쉴드 부메랑 5^000000",
"계열 : ^993300액티브^000000",
"타입 : ^777777원거리 물리^000000",
"대상 : ^777777대상 1체^000000",
"내용 : ^777777방패를 장착한 상태에서만 사용가능한 스킬로서 방패로 적을 공격한다.",
"캐릭터의 명중율에 영향을 받으며, 시전자의 베이스 레벨, 방패의 제련도 및 무게로 인한 추가 데미지가 존재한다.^000000",
"[Lv 1] : ^777777ATK 500% , 명중률 +20^000000",
"[Lv 2] : ^777777ATK 700% , 명중률 +20^000000",
"[Lv 3] : ^777777ATK 900% , 명중률 +20^000000",
"[Lv 4] : ^777777ATK 1100%, 명중률 +20^000000",
"[Lv 5] : ^777777ATK 1300% , 명중률 +20^000000"
```

**Comment by Lemongrass3110** [comment]
Fixed in 382d6bf
